rustls-cert-gen/src/args.rs

```diff
@@ -0,0 +1,95 @@
+//! Command Line argument parsing
+#![allow(missing_docs)]
+
+use std::{net::IpAddr, path::PathBuf};
+
+use bpaf::Bpaf;
+use rcgen::SanType;
+
+#[derive(Debug, Clone, Bpaf)]
+#[bpaf(fallback(Purpose::Omit))]
+/// Extended Key Usage Purpose (rfc5280, section 4.2.1.12)
+pub enum Purpose {
+	/// Extended Key Usage Purpose: ClientAuth
+	ClientAuth,
+	/// Extended Key Usage Purpose: ServerAuth
+	ServerAuth,
+	#[bpaf(skip)]
+	/// Default case is to omit adding Extended Key Usage Purpose
+	Omit,
+}
+
+#[derive(Clone, Debug, Bpaf)]
+#[bpaf(options)]
+/// rustls-cert-gen Tls Certificate Generator
+pub struct Options {
+	/// Output Directory for generated files
+	#[bpaf(short, long, argument("output/path/"))]
+	pub output: PathBuf,
+	/// Persist ca signing-key to `output` dir
+	#[bpaf(long, switch)]
+	pub insecure: bool,
```

Comment:
I'm not sure about calling this mode "insecure" or not doing it by default (IMO it should by default emit all files, including the signing key)

Reply:
I am happy to change it, but I would like to share my thought process first. I think the signing key is only useful for "advanced" usage. Namely, as input to another tool (or a possible future iteration of this one)  to sign more certificates. Someone looking for a quick way to generate certificates for a TLS connection won't not need it. I think such a person might also be less likely to have a complete grasp of the why and wherefore of all the files. They might also be less informed about secure practices for key management and the implications of leaking the signing key  of their ad-hoc certificate-authority. That's how I came to naming the switch `insecure`. So the idea was to make it easy for those that want something easy and provide a path to more advanced usage for those that know what they are doing.

Reply:
I don't like the name here because it doesn't feel like persisting a private key to disk is inherently "insecure". Unless we have functionality for generating new certificates that are signed with an existing on-disk CA private key, I'm not sure it makes sense to persist the CA private key to disk because it'll be relatively hard to use for anything. Adding such functionality (to sign with a persisted CA private key) makes sense to me and probably shouldn't be too hard, but also probably doesn't need to be in this PR.

Reply:
IMO it makes sense to have the key around even if such a functionality doesn't exist yet, as then the key is around once we add that functionality, instead of being lost.

And yeah, full agree, persisting a private key to disk is not inherently insecure. This is a tool in the end, and yeah if you deploy the CA into a root store then the private key becomes very powerful. But that's up to the users.

Reply:
OK `--secure` switch was removed in latest commit and all files will always be persisted.